dataland-frontend/src/components/forms/parts/fields/NonNegativeNumberFormField.vue

```diff
@@ -0,0 +1,26 @@
+<template>
+  <div class="form-field">
+    <UploadFormHeader :name="displayName" :explanation="info" :is-required="required" />
+    <FormKit
+      type="text"
+      :name="name"
+      :validation-label="validationLabel ?? displayName"
+      :validation="`number|min:0|${validation}`"
+      :placeholder="placeholder"
+      inner-class="short"
+    />
+  </div>
+</template>
+
+<script lang="ts">
+import UploadFormHeader from "@/components/forms/parts/elements/basic/UploadFormHeader.vue";
+import { defineComponent } from "vue";
+import { FormKit } from "@formkit/vue";
+import { FormFieldPropsWithPlaceholder } from "@/components/forms/parts/fields/FormFieldProps";
+
+export default defineComponent({
+  name: "NonNegativeNumberFormField",
+  components: { FormKit, UploadFormHeader },
+  props: FormFieldPropsWithPlaceholder,
+});
+</script>
```

Comment:
Should be possible. Will take care of it

Reply:
There are more generic field components that are almost copies of each other.
Maybe we should go through them together and think about the same approach there:
Instead of two seperate components, just use one, and adjust the "validation" or "name" based on the use case.

Reply:
E.g. `FreeTextFormField` and `InputTextFormField` have only one diff:   One has `type` set to "text", and the other one to "textarea".
If we create a generic `TextFormField` and pass the type as prop, we could avoid this.

Reply:
Fixed

Reply:
The text form fields cannot be easily united, since the `type` prop cannot be passed in the `v-for <component/>`.
Therefore we leave it as it is.

We could not find another pair of components, that could be united to a more generic component.

Therefore I close this.

Reply:
What do you mean by the `type` prop cannot be passed in the `<component/>`? It sure would need to be added in the lksg converter, but why would that not be possible. I prefer the current approach though.

Reply:
Sorry I did not describe it precisely.
Yes, `type` could be added, and then we could introduce a generic Text-component, but Marc had some good reasons why it might be better to not do it at the moment.